code/game/machinery/dialysis.dm

```diff
@@ -0,0 +1,91 @@
+/obj/machinery/dialysis
+	name = "dialysis machine"
+	desc = "A machine used to purge reagents from the blood. Simply connect a person to it, turn it on, and wait for the ping."
+	icon = 'icons/obj/stationobjs.dmi'
+	icon_state = "rand_machine_1"
+	idle_power_usage = 150
+	active_power_usage = 450
+	anchored = 1
+	density = 1
+	var/list/connections = list()
+	var/max_connections = 1
+	var/reagent_removal_rate = 5
+
+	machine_flags = SCREWTOGGLE | CROWDESTROY | WRENCHMOVE
+
+/obj/machinery/dialysis/New()
+	. = ..()
+
+	component_parts = newlist(
+		/obj/item/weapon/circuitboard/dialysis,
+		/obj/item/weapon/stock_parts/manipulator,
+		/obj/item/weapon/stock_parts/manipulator,
+		/obj/item/weapon/stock_parts/manipulator,
+		/obj/item/weapon/stock_parts/micro_laser,
+		/obj/item/weapon/stock_parts/micro_laser,
+		/obj/item/weapon/stock_parts/console_screen
+	)
+
+	RefreshParts()
+
+/obj/machinery/dialysis/RefreshParts()
+	var/lasercount = 1
+	var/manipcount = 1
+	for(var/obj/item/weapon/stock_parts/SP in component_parts)
+		if(istype(SP, /obj/item/weapon/stock_parts/micro_laser))
+			lasercount += SP.rating-1
+		if(istype(SP, /obj/item/weapon/stock_parts/manipulator))
+			manipcount += SP.rating-1
+	max_connections = initial(max_connections) * manipcount
+	reagent_removal_rate = initial(reagent_removal_rate) * lasercount
+
+/obj/machinery/dialysis/Destroy()
+	connections.Cut()
+	..()
+
+/obj/machinery/dialysis/MouseDropFrom(over_object, src_location, over_location)
+	if(isobserver(usr))
+		return ..()
+	if(usr.incapacitated())
+		return ..()
+	if(isanimal(usr))
+		return ..()
+	if(!usr.Adjacent(src))
+		return ..()
+
+	if(connections.Find(over_object))
+		visible_message("[over_object] is detached from \the [src]")
+		connections.Remove(over_object)
+		return
+
+	if(ismob(over_object) && Adjacent(over_object))
+		if(ishuman(over_object))
+			var/mob/living/carbon/human/H = over_object
+			if(H.species && (H.species.chem_flags & NO_INJECT))
+				H.visible_message("<span class='warning'>[usr] struggles to place the IV into [H] but fails.</span>","<span class='notice'>[usr] tries to place the IV into your arm but is unable to.</span>")
+				return
+
+		if(connections.len > max_connections)
```

Comment:
Well, it's not really `max_connections` if you're allowed to go over by 1
I think you meant to use `>=`